File: ruby/raindrops/raindrops.rb
class Raindrops
  FACTORS = { 3 => "Pling", 5 => "Plang", 7 => "Plong"}

  def self.convert(num)
    result = FACTORS.each_with_object("") do |(f, s), out|
      out << s if num % f == 0 
    end

    result.empty? ? num.to_s : result
  end
end

File: scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  val factors = Map(
    3 -> "Pling",
    5 -> "Plang",
    7 -> "Plong"
  )

  def convert(n: Int): String = {
    val pieces = factors.filterKeys(n % _ == 0).values
    if (pieces.isEmpty)
      n.toString
    else
      pieces.mkString("")
  }
}

